src/messages/Image.hpp

```diff
@@ -80,13 +85,57 @@ class Image : public std::enable_shared_from_this<Image>, boost::noncopyable
 
     void setPixmap(const QPixmap &pixmap);
     void actuallyLoad();
+    void expireFrames();
 
     const Url url_{};
     const qreal scale_{1};
     std::atomic_bool empty_{false};
 
-    // gui thread only
+    mutable std::chrono::time_point<std::chrono::steady_clock> lastUsed_;
+
     bool shouldLoad_{false};
+
+    // gui thread only
     std::unique_ptr<detail::Frames> frames_{};
+
+    friend class ImageExpirationPool;
+};
+
+class ImageExpirationPool
+{
+private:
+    friend class Image;
+
+    ImageExpirationPool();
+    static ImageExpirationPool &instance();
+
+    /**
+     * @brief Stores a reference to the given Image in the ImagePool.
+     * 
+     * The caller is responsible that the pointer will be valid for as long as
```

Comment:
Please remove these doc comments. The description given about "The caller is responsible that the pointer" is already incorrect. The function itself is self explanatory.